pybamm/simulation.py

```diff
@@ -768,12 +769,19 @@ def solve(
 
                     logs["termination"] = step_solution.termination
                     # Only allow events specified by experiment
-                    if not (
+                    if (
+                        step_termination == "final time"
+                        and step.uses_default_duration is True
+                    ):
+                        callbacks.on_experiment_infeasible_time(logs)
+                        feasible = False
+                        break
+                    elif not (
                         isinstance(step_solution, pybamm.EmptySolution)
                         or step_termination == "final time"
                         or "[experiment]" in step_termination
                     ):
-                        callbacks.on_experiment_infeasible(logs)
+                        callbacks.on_experiment_infeasible_event(logs)
                         feasible = False
                         break
```

Comment:
It is usually good to end the chains in an else of some sort to be clearer for future devs. A quick workaround is to just break up the last elif:
```python
elif condition:
    code()
```
becomes
```python
else:
    if condition:
        code()
```

Not the prettiest, but it is clearer that the chain is done.